core/modules/replica/tools/qserv-replica-job-verify.cc

```diff
@@ -0,0 +1,170 @@
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/// qserv-replica-job-verify.cc is a single job Controller application
+/// which is meant to run the corresponding job.
+
+// System headers
+#include <atomic>
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+
+// Qserv headers
+#include "proto/replication.pb.h"
+#include "replica/Controller.h"
+#include "replica/VerifyJob.h"
+#include "replica/ReplicaInfo.h"
+#include "replica/ServiceProvider.h"
+#include "util/BlockPost.h"
+#include "util/CmdLineParser.h"
+
+using namespace lsst::qserv;
+
+namespace {
+
+// Command line parameters
+
+std::string configUrl;
+size_t      maxReplicas;
+bool        computeCheckSum;
+bool        progressReport;
+bool        errorReport;
+bool        detailedReport;
+
+/// Run the test
+bool run() {
+
+    try {
+
+        ///////////////////////////////////////////////////////////////////////
+        // Start the controller in its own thread before injecting any requests
+        // Note that omFinish callbak which are activated upon a completion
+        // of the requsts will be run in that Controller's thread.
+
+        replica::ServiceProvider::Ptr const provider   = replica::ServiceProvider::create(configUrl);
+        replica::Controller::Ptr      const controller = replica::Controller::create(provider);
+
+        controller->run();
+
+        ////////////////////////////////////////
+        // Find all replicas accross all workers
+
+        std::atomic<bool> finished{false};
+        auto job = replica::VerifyJob::create (
+            controller,
+            std::string(),
+            [&finished] (replica::VerifyJob::Ptr const& job) {
+                finished = true;
+            },
+            [] (replica::VerifyJob::Ptr const& job,
+                replica::ReplicaDiff const& selfReplicaDiff,
+                std::vector<replica::ReplicaDiff> const& otherReplicaDiff) {
```

Comment:
I think typically the indenting on the wrapped function parameters is done at 8 chars, to disambiguate the body of the function, which is indented 4 chars. With this many lines in this context I think it's worth making the difference obvious somehow.

Reply:
Good point! Fixed